File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/server/messages/ServerMessageOpenDocument.java
package net.stzups.board.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.Document;
import net.stzups.board.data.objects.canvas.Canvas;
import net.stzups.board.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageOpenDocument extends ServerMessage {
    private Document document;
    private Canvas canvas;

    public ServerMessageOpenDocument(Document document, Canvas canvas) {
        super(ServerMessageType.OPEN_DOCUMENT);
        this.document = document;
        this.canvas = canvas;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeLong(document.getId());
        this.canvas.serialize(byteBuf);
    }
}

File: app/frontend/src/scripts/common/protocol/server/messages/ServerMessageOpenDocument.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import {Canvas} from "../../../canvas/Canvas.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageOpenDocument extends ServerMessage {
    id: bigint;
    canvas: Canvas;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.id = byteBuffer.readBigInt64();
        this.canvas = new Canvas(byteBuffer);
    }

    getType(): ServerMessageType {
        return ServerMessageType.OPEN_DOCUMENT;
    }
}